project-5/src/components/MemeInput.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from "react";
+
+const MemeInput = () => {
+  const [meme, setMeme] = useState({
+    topText: "",
+    bottomText: "",
+    randomImg: "http://i.imgflip.com/1bij.jpg",
+  });
+
+  const { topText, bottomText, randomImg } = meme;
+  const [allMemeImgs, setAllMemeImgs] = useState([]);
+
+  useEffect(function () {
+    fetch("https://api.imgflip.com/get_memes")
+      .then((response) => response.json())
+      .then((api) => setAllMemeImgs(api.data.memes));
```

Comment:
- [ ] Its better to use `async - await` method instead of using promises.
```
const fetchData = async () => {
    const { data } = await fetch("https://api.imgflip.com/get_memes");
    setAllMemeImgs(data.memes);
}

useEffect(() => {
    fetchData()
}, [])
```